src/Results.types.tsx

```diff
@@ -0,0 +1,10 @@
+import type { Author, ResponseSearch } from './Types';
+
+export interface ResultsData {
+    title: string;
+    data: ResponseSearch;
+    authors: Author[];
+    onClick: (e: React.PointerEvent<HTMLLIElement>) => void;
```

Comment:
I think you overcomplicated this a bit. Basically I would go only for one callback `onPageChange` instead those three - `onClick`, `onPageUp` and `onPageDown` - and then inside `Pagination` component you can go for `props.onClick(value)` to pass to callback specific page that user wants to go to.